src/js/UserList.js

```diff
@@ -0,0 +1,67 @@
+
+export default function UserList({onSelect, onAdd, onDelete}) {
+  const $userList = document.querySelector('#user-list');
+  
+  let $state = {};
+  
+  this.setState = (newState) => {
+    $state = newState;
+    this.render();
+  }
+
+  this.render = () => {
+    $userList.insertAdjacentHTML('beforeend',this.template());
+    this.setEvent();
+  }
+  
+  this.template = () => {
+    return `
+      <button class="ripple user-create-button" data-action="createUser">
+        + 유저 생성
+      </button>
+      <button class="ripple user-delete-button" data-action="deleteUser">
+        삭제 -
+      </button>`;
+  }
+  this.setEvent = () => {
+    
+    this.addEvent1();
+    this.addEvent2();
+    this.addEvent3();
+  } 
+
+
+  this.addEvent1 = () => {
+    const $userListBtn = document.querySelectorAll('.ripple');
+    
+    [...$userListBtn].map(btn => btn.addEventListener('click', event => {
+    if (!event.target.matches('.ripple')) return;
+    if (!event.target.hasAttribute('id')) return;
+    
+    const selectedId = event.target.id;
+    onSelect(selectedId);
+    }))
+  }
```

Comment:
```suggestion
    [...$userListBtn].map(btn => btn.addEventListener('click', ({ target })=> {
    if (!target.matches('.ripple')) return;
    if (!target.hasAttribute('id')) return;
    
    const selectedId = target.id;
    onSelect(selectedId);
    }))
  }
```

저도 저번주에 다른 분께 배웠는데 이런 식으로 사용하면 event 중복으로 사용하는 걸 줄여준다고 해요..!! 👍👍

Reply:
구조분해할당을 이용하는 건가요?? 한번 알아봐야겠네요 감사합니다!